hudi-utilities/src/main/java/org/apache/hudi/utilities/transform/ChainedTransformer.java

```
@@ -109,6 +123,29 @@ private void validateIdentifier(String id, Set<String> identifiers, String confi
     }
   }
 
+  private Option<Schema> validateAndGetTransformedSchema(TransformerInfo transformerInfo, Dataset<Row> dataset, Option<Schema> incomingSchemaOpt,
+                                                         JavaSparkContext jsc, SparkSession sparkSession, TypedProperties properties) {
+    if (!incomingSchemaOpt.isPresent()) {
+      throw new HoodieSchemaException(String.format("Source schema not available for transformer %s", transformerInfo));
+    }
+    Schema incomingSchema = incomingSchemaOpt.get();
+
+    // Get expected target schema from transformer and actual target schema from struct
+    Schema targetSchema = AvroConversionUtils.convertStructTypeToAvroSchema(dataset.schema(), incomingSchema.getName(),
+        incomingSchema.getNamespace());
+    Option<Schema> expectedTargetSchemaOpt = transformerInfo.getTransformer().transformedSchema(jsc, sparkSession, incomingSchema, properties);
+    if (!expectedTargetSchemaOpt.isPresent()) {
```

we can move this to L 130 may be. just to fail fast.